static/app/components/copyToClipboardButton.tsx

```diff
@@ -0,0 +1,69 @@
+import {ComponentProps, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button, ButtonProps} from 'sentry/components/button';
+import Clipboard from 'sentry/components/clipboard';
+import {IconCopy} from 'sentry/icons';
+import {t} from 'sentry/locale';
+
+type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;
+
+type Props = {
+  text: string;
+  iconSize?: ComponentProps<typeof IconCopy>['size'];
+} & Overwrite<
```

Comment:
wtf is Overwrite 👀

Reply:
lets use interface extends here?

Reply:
computer says no :(

<img width="884" alt="SCR-20230505-nngn" src="https://user-images.githubusercontent.com/187460/236577502-b5a963fd-f6ba-4e15-853c-7c055b066417.png">

Reply:
how dare they